.travis.yml

```diff
@@ -20,166 +17,144 @@ sudo: false
+      - SETUP_XVFB=True
+      - EVENT_TYPE='push pull_request'
 
 stages:
-   # Do the style check and a single test job, don't proceed if it fails
-   - name: Initial tests
-   # Do the rest of the tests
-   - name: Comprehensive tests
-   - name: Cron tests
-     if: type = cron
+    # Do the style check and a single test job, don't proceed if it fails
+    - name: Initial tests
+    # Do the rest of the tests
+    - name: Comprehensive tests
+    - name: Cron tests
+      if: type = cron
 
 matrix:
 
     # Don't wait for allowed failures
     fast_finish: true
 
     include:
+
         # Try MacOS X. Use a slightly old numpy version to help test against
         # all supported numpy versions.
-        - os: osx
-          stage: Cron tests
-          env: SETUP_CMD='test --remote-data=astropy'
-               CONDA_DEPENDENCIES="$CONDA_ALL_DEPENDENCIES clang"
-               PIP_DEPENDENCIES='jplephem asdf'
-               CCOMPILER=clang
-               EVENT_TYPE='cron'
+        # - os: osx
+        #   python: 3.6
+        #   # stage: Cron tests
+        #   env: TOXARGS='--remote-data=astropy'
+        #        CCOMPILER=clang
+        #        # CONDA_DEPENDENCIES="$CONDA_ALL_DEPENDENCIES clang"
+        #        # PIP_DEPENDENCIES='asdf'
+        #        # EVENT_TYPE='cron'
 
         # Try all python versions and Numpy versions. Since we can assume that
         # the Numpy developers have taken care of testing Numpy with different
         # versions of Python, we can vary Python and Numpy versions at the same
         # time.
-        # Run this test using native pytest
         - os: linux
-          env: PYTHON_VERSION=3.5 NUMPY_VERSION=1.13
-               INSTALL_CMD='python setup.py build_ext --inplace'
-               PIP_DEPENDENCIES='pytest-astropy'
-               TEST_CMD='pytest --open-files --doctest-rst'
-               PYTEST_VERSION=3.7
-          script:
-            - $INSTALL_CMD
-            - $TEST_CMD
+          python: 3.5
+          env: TOXENV=py35
+               TOXOPTS='--develop --force-dep \'pytest==3.7\''
+               TOXARGS='--cov=astropy --open-files --doctest-rst'
+               CFLAGS='--coverage -fno-inline-functions -O0'
 
         # Now try with all optional dependencies. We also include the --readonly
         # flag to make sure no files are being written to the temporary install
         # location during testing.
         - os: linux
-          env: PYTHON_VERSION=3.5 SETUP_CMD='test --remote-data=astropy --readonly'
-               CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="jplephem asdf"
-               LC_CTYPE=C.ascii LC_ALL=C
-               PYTEST_VERSION=3.7
+          python: 3.5
+          env: TOXENV=py35-alldeps
+               TOXOPTS='--force-dep \'pytest==3.7\''
+               TOXARGS='--cov=astropy --remote-data=astropy'
+               CFLAGS='--coverage -fno-inline-functions -O0'
```

Comment:
I wonder if it might be better to just put all of this into a `tox` environment. See what I've done for the `asdf` config:
```ini
[testenv:coverage]
basepython= python3.7
deps=
    gwcs
conda_deps=
    pytest<=3.6
    pytest-astropy
    astropy
    coverage
    coveralls
commands=
    coverage run --source=asdf -m pytest --remote-data --open-files
    coverage report -m
    coveralls --rcfile='asdf/tests/coveragerc'
passenv= TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH
```

I admit my case is slightly different since I want to avoid using the Astropy test runner.

Reply:
Actually I was thinking about doing coverage by default in tox, but it may be annoying to always have it.
I usually use pytest-cov which is simpler than coverage, is there any benefit in using the latter ?

Reply:
I think for the core astropy package we shouldn't have coverage by default as it does have a performance impact. I think to facilitate the review process by other core developers, we should avoid changing too much about the default builds at the same time as switching to tox, so it's probably best to keep the same conceptual builds as currently and then submit follow-up PRs to improve the configuration.